.ci-builds/.azure-pipelines.yml

```diff
@@ -159,31 +136,79 @@ jobs:
         CXX: clang++
         LANG: "en_US@calendar=gregorian;hours=h12"
 #-------------------------------------------------------------------------
-- job: ICU4C_Clang13_Cpp20_Ubuntu_2004
-  displayName: 'C: Linux Clang-13 C++20 (Ubuntu 20.04)'
+- job: ICU4C_Clang14_Ubuntu_2004_WarningsAsErrors
+  displayName: 'C: Linux Clang-14 WarningsAsErrors (Ubuntu 20.04)'
+  timeoutInMinutes: 30
+  pool:
+    vmImage: 'ubuntu-20.04'
+  steps:
+    - checkout: self
+      lfs: true
+      fetchDepth: 10
+    # Install Clang-14 from https://apt.llvm.org/
+    - script: |
+        curl -Ls https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+        sudo apt-add-repository 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main'
+        sudo apt update
+        sudo apt install -qy clang-14 lld-14 libc++-14-dev libc++abi-14-dev
+      displayName: 'Install Clang-14'
+    - script: |
+        export CPPFLAGS="-Werror -Wall -Wextra -Wextra-semi -Wundef -Wnon-virtual-dtor -Wsuggest-override" && cd icu4c/source && ./runConfigureICU Linux && make -j2 tests
+      displayName: 'Build only (WarningsAsErrors)'
+      env:
+        CC: clang-14
+        CXX: clang++-14
+#-------------------------------------------------------------------------
+- job: ICU4C_Clang14_Cpp20_Ubuntu_2004
+  displayName: 'C: Linux Clang-14 C++20 (Ubuntu 20.04)'
   timeoutInMinutes: 30
   pool:
     vmImage: 'ubuntu-20.04'
   steps:
     - checkout: self
       lfs: true
       fetchDepth: 10
-    # Install Clang-13 from https://apt.llvm.org/
+    # Install Clang-14 from https://apt.llvm.org/
     - script: |
         curl -Ls https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
-        sudo apt-add-repository 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-13 main'
+        sudo apt-add-repository 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main'
         sudo apt update
-        sudo apt install -qy clang-13 lld-13 libc++-13-dev libc++abi-13-dev
-      displayName: 'Install Clang-13'
+        sudo apt install -qy clang-14 lld-14 libc++-14-dev libc++abi-14-dev
+      displayName: 'Install Clang-14'
     - script: |
         cd icu4c/source && ./runConfigureICU Linux && make -j2 check
       displayName: 'Build and Test'
       env:
-        CC: clang-13
+        CC: clang-14
         CPPFLAGS: '-Werror'
-        CXX: clang++-13
+        CXX: clang++-14
         CXXFLAGS: '-std=c++20'
 #-------------------------------------------------------------------------
+- job: ICU4C_Clang14_Cpp20_libcpp_Ubuntu_2004
```

Comment:
optional/FYI: My inclination is to change the existing C++20 clang14 bot to libc++ and leave the C++20 gcc11 bot with the GNU library... but this is fine, too.

Reply:
I thought it might be interesting to have both, since they (mostly) run in parallel. (We'd have both "Clang-14 with GNU libstdc++" and "Clang-14 with LLVM libc++"). But can remove it if you'd prefer.

Reply:
I don't care strongly, hence the approval :-)